### PR TITLE
S4-214: Added piwik tracking to hyperlinks

### PR DESCRIPTION
**Pull request by faggc**
## Description

Added a goal trigger for opening the PDF link on the Confirmation page

Added event ids to relevant hyperlinks

- Confirmation page guidance
- Who to tell guidance
- The change link on the check your answers page

## Coding Standards

Code Style Guide: https://github.com/companieshouse/dissolution-web/wiki/Code-Style-Guide

## Checklist

- [x] 3 Amigos
- [ ] Acceptance Criteria met
- [x] Branch named {feature|hotfix|task}/{S4-*}
- [x] Commit messages are meaningful
- [x] Manually tested
- [x] All unit tests passing
- [ ] WAVE accessibility testing tool ran on new or updated pages
- [x] Pulled latest master into feature branch
- [ ] Code reviewed
- [ ] New Docker environment variables are consistent with those on the Rebel1 environment
- [ ] If your pull request depends on any other, please link them in the description

## Screenshots of new or updated views
Added ids:
- check-answers-change-email
- confirmation-read-guidance
- who-to-tell-read-guidance
Screenshot 2020-09-14 at 13 35 37

### CODE REVIEW

File: src/views/check-your-answers.njk [hunk 1]
```diff
@@ -23,7 +23,7 @@
         {{
           govukSummaryList({
             classes: 'govuk-!-margin-bottom-0',
-            attributes: { id: 'director-details-' + loop.index0 },
+            attributes: { id: 'director-details' },
```
Comment (peadarkelly) [inline src/views/check-your-answers.njk:26]: This will cause accessibility tests to fail. If 2 directors are selected, then a duplicate ID will be present on the page.

File: src/views/view-final-confirmation.njk [hunk 2]
```diff
@@ -4,7 +4,7 @@
 {% from "govuk/components/warning-text/macro.njk" import govukWarningText %}
 
 {% block pageTitle %}
-  View final confirmation
+  View final information
```
Comment (peadarkelly) [inline src/views/view-final-confirmation.njk:7]: why was this changed? Is this intentional?

File: src/views/view-final-confirmation.njk [hunk 3]
```diff
@@ -22,12 +22,12 @@
 
       <h2 class="govuk-heading-m">What to do next</h2>
 
-      <p class="govuk-body"><a href="{{ Paths.CERTIFICATE_DOWNLOAD_URI }}" target="_blank" class="govuk-link">Download the signed copy of the application (opens in a new tab)</a>.</p>
+      <p class="govuk-body"><a href="#" target="_blank" onclick="javascript:_paq.push(['trackGoal', '{{ piwik.PIWIK_CONFIRMATION_PAGE_PDF_GOAL_ID }}');" class="govuk-link">Download the signed copy of the application (PDF 238Kb) (opens in a new tab)</a>.</p>
```
Comment (peadarkelly) [inline src/views/view-final-confirmation.njk:25]: Why have you changed the href? This hyperlink currently does nothing. Also why are you adding the (PDF 228kb) text? This story is just about piwik, there's nothing in the ACs about updating content. Was this called out at 3 amigos or something?

File: src/models/piwikConfig.ts [hunk 4]
```diff
@@ -2,4 +2,5 @@ export default interface PiwikConfig {
   url: string
   siteId: string
   landingPageStartGoalId: string
+  confirmationPagePDFGoalId: string
```
Comment (peadarkelly) [inline src/models/piwikConfig.ts:5]: shouldn't these be `number`s? CC @matthewdoreilly

**Review by peadarkelly — CHANGES_REQUESTED** [review]
Seems to be some unnecessary content changes made. Can you also attach screenshots of the updates in Matomo to show the events being sent correctly?

File: src/views/view-final-confirmation.njk (hunk 2 above)
Comment (faggc) [reply src/views/view-final-confirmation.njk:7]: No, I'm not sure why this changed

File: src/views/check-your-answers.njk (hunk 1 above)
Comment (faggc) [reply src/views/check-your-answers.njk:26]: This should not have changed

File: src/views/view-final-confirmation.njk (hunk 3 above)
Comment (faggc) [reply src/views/view-final-confirmation.njk:25]: I have no idea why this is appearing, editing it out

File: src/models/piwikConfig.ts (hunk 4 above)
Comment (matthewdoreilly) [reply src/models/piwikConfig.ts]: Yep, both `siteId` and `landingPageStartGoalId` should have been `number`.
Would you be able to update these types @faggc?

Comment (faggc) [reply src/models/piwikConfig.ts]: Sure thing

**Comment by faggc** [comment]
Removed all code related to the event until it is created.

**Comment by peadarkelly** [comment]
@faggc can you attach screenshots of Matomo regarding how these events are displayed? Just so that I can confirm that the naming is consistent.

**Review by peadarkelly — APPROVED** [review]
